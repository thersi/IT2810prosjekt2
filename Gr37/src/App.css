.App {
  text-align: center;
}



.wrapper {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-auto-rows: minmax(70px, auto);
  gap: 10px;
}


.toggle {
  grid-row: 2;
  grid-column: 1 / 3;
  display: flex;
  align-items: center;
  justify-content: left;
  padding-left: 7%;
}

.displayChart {
  padding-top: 7%;
  min-height: 400px;
  min-width: 300px;
  grid-column: 2 / 7;
  grid-row: 4 ;
  position: relative;
  padding-right: 40px;

}

.header {
  grid-column: 1/8;
  grid-row: 1;
  padding-top: 1%;
  padding-bottom: 1%;
  background-color: rgb(51, 51, 51, 0.1);
  font-size: 1.5rem;

}

.search {
  grid-column: 1/8;
  grid-row: 3;
}

@media screen and (min-width:375px){
  .displayChart {
    grid-column: 0 / 8; 
  }
}

@media only screen and (max-width: 600px) {
  .toggle {
    grid-row: 2;
    grid-column: 1 / 8;
    padding-left: 0%;
    align-items: center;
    justify-content: center;
  }
  .displayChart {
    padding-left: 0px;
    padding-right: 5%;
    grid-column: 1 / 8;
  }
}

