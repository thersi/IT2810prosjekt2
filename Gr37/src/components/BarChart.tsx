import { Bar } from 'react-chartjs-2'

/* 
Component ment to show number of commits per person 
*/

export default function BarChart({ xAxis, yAxis }: { xAxis: Array<string>; yAxis: Array<number> }) {

    //Need to take in arrays of names and commits (same order)

    const data = {
        labels: xAxis,
        datasets: [{
            label: 'Number of commits',
            data: yAxis,
            backgroundColor: [
                'rgba(255, 99, 132, 0.2)',
                'rgba(54, 162, 235, 0.2)',
                'rgba(255, 206, 86, 0.2)',
                'rgba(75, 192, 192, 0.2)',
                'rgba(153, 102, 255, 0.2)',
                'rgba(255, 159, 64, 0.2)'],
            borderColor: [
                'rgba(255, 99, 132, 1)',
                'rgba(54, 162, 235, 1)',
                'rgba(255, 206, 86, 1)',
                'rgba(75, 192, 192, 1)',
                'rgba(153, 102, 255, 1)',
                'rgba(255, 159, 64, 1)'],
            borderWidth: 1
        }],
        title: {
            text: "Number of commits per person",
            fontSize: 25
        }
    }
    const options = {
        maintainAspectRatio: false,

        plugins: {
            title: {    
                display: true,
                text: "Number of commits per person",
                padding: {
                    top: 20,
                    bottom: 25
                },
                font: {
                    size: 20,
                },
            }
        },
        scales: {
            y: {
                beginAtZero: true,
                title: {
                    display: true,
                    text: 'Commits',
                    padding: {
                        top: 5,
                        bottom: 5
                    },
                }
            },
            x: {
                title: {
                    display: true,
                    text: 'Group members',
                    padding: {
                        top: 5,
                        bottom: 25
                    },
                }
            }
        }
    }
    return (
        
            <Bar id="barChart" data={data} options={options} />
        
    )
}
