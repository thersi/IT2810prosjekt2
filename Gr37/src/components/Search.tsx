import React, { useState } from "react"
import { Card, Select, MenuItem, SelectChangeEvent, TextField, Button} from '@mui/material';
import AdapterDateFns from '@mui/lab/AdapterDateFns';
import { LocalizationProvider, DatePicker } from '@mui/lab';

export default function Search() {
    const [value, setValue] = useState('');
    const [fromDate, setFromDate] = useState<Date | null>(new Date())
    const [toDate, setToDate] = useState<Date | null>(new Date());


    const handleValueChange = (event: SelectChangeEvent) => {
        setValue(event.target.value as string);
    }

    const handleFromDateChange = (newValue: Date | null) => {
        if (newValue != null) {
            if (toDate != null && newValue.getTime() <= toDate.getTime()) { 
                setFromDate(newValue);
            }
        console.log(fromDate)
    }
    }

    const handleToDateChange = (newValue: Date | null) => {
        if (newValue != null) {
            if (fromDate != null && newValue.getTime() >= fromDate.getTime()) { 
                setToDate(newValue);
            }
        console.log(toDate)
    }
    }

    const onSearch = () => {
        console.log(fromDate, toDate, value)
    }

    return (
        <Card>
            <Select
                labelId="demo-simple-select-label"
                id="demo-simple-select"
                value={value}
                label="Age"
                onChange={handleValueChange}
            >
                <MenuItem value={'Issues'}>Issues</MenuItem>
                <MenuItem value={'Commits'}>Commits</MenuItem>
            </Select>
            
            <LocalizationProvider dateAdapter={AdapterDateFns}>
                <DatePicker
                    label="From"
                    value={fromDate}
                    onChange={handleFromDateChange}
                    renderInput={(params) => <TextField {...params} />}
                />
                <DatePicker
                    label="To"
                    value={toDate}
                    onChange={handleToDateChange}
                    renderInput={(params) => <TextField {...params} />}
                />
            </LocalizationProvider>
            <Button variant="contained" onClick={onSearch}>Search</Button>
        </Card>

    )

}