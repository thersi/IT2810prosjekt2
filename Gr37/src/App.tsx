import React, { useContext } from "react";
import './App.css';
import Search from './components/Search'
import BarChart from './components/BarChart'
import LineChart from './components/LineChart'
import { ThemeCtx} from './ThemeProvider'



function App() {
  const { theme, toggleTheme } = useContext(ThemeCtx);
  let bar = false; /*implement means of setting the condition
  based on the input to the search component*/


  const renderCorrectChart = () => {
    if (bar) {
      return <BarChart
      xAxis={['Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange']}
      yAxis={[12, 19, 3, 5, 2, 3]}
    />
    } else {
      return <LineChart
      dates={['Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange']}
      merges={[12, 19, 3, 5, 2, 3]}
    />;
    }
  }

  return (
      <div className="App">
        <button id= 'toggleTheme' onClick={toggleTheme}>
        Switch to {theme === "light" ? "dark" : "light"} mode
      </button>
        <Search></Search>
        <div className="displayChart">
        {renderCorrectChart()}
        </div>
      </div>
  );
}

export default App;
