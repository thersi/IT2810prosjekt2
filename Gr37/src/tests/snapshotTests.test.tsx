import React from 'react'
import ReactDOM from 'react-dom'
import Search from '../components/Search'
import BarChart from '../components/BarChart'
import App from '../App'
import { cleanup, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup);
let div: HTMLDivElement;



it('Search renders without crashing', () => {
    div = document.createElement('div');
    ReactDOM.render(<Search></Search>, div)

})

it('matches snapshot', () => {
    const tree = renderer.create(<Search></Search>).toJSON()
    expect(tree).toMatchSnapshot();
})

it('matches snapshot', () => {
    const tree = renderer.create(<App/>).toJSON()
    expect(tree).toMatchSnapshot();
})

