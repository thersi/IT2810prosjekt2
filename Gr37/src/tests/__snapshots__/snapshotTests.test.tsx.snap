// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="wrapper"
  >
    <header
      className="header"
    >
      Repository data for group 37
    </header>
    <div
      className="search"
    >
      <div>
        Loading....
      </div>
    </div>
    <div
      className="toggle"
    >
      <button
        className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard css-1x4vipr-MuiButtonBase-root-MuiToggleButton-root"
        disabled={false}
        id="toggleTheme"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
        value="web"
      >
        light
         mode
      </button>
    </div>
    <div
      className="displayChart"
    >
      <p>
        Perform a search to display data
      </p>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 2`] = `
<canvas
  data-testid="canvas"
  height={150}
  id="barChart"
  onClick={[Function]}
  role="img"
  width={300}
/>
`;

exports[`renders linechart correctly 1`] = `[Function]`;
